pkg/platform/kube/functionres/lazy.go

```diff
@@ -1416,8 +1416,6 @@ func (lc *lazyClient) populateServiceSpec(functionLabels labels.Set,
 		}
 		if serviceTypeIsNodePort {
```

Comment:
perhaps you need to `switch` on `spec.Type` and assign a port in case of `v1.ServiceTypeNodePort`, do nothing in case of `v1.ServiceTypeClusterIp` and default to 0 to preserve the current behaviour? 
Because, afaik, a service type might can loadBalancer and that change would affect it